html/js/product-multilingual.js

```diff
@@ -494,8 +494,112 @@ function update_display(imagefield, first_display) {
 	$(document).foundation('equalizer', 'reflow');
 }
 
+function initializeTagifyInputs() {
+  const inputs = document.querySelectorAll('input.tagify-me');
+  for (let i = 0; i < inputs.length; ++i) {
+    initializeTagifyInput(inputs[i]);
+  }
+}
+
+const maximumRecentEntriesPerTag = 3;
+function initializeTagifyInput(el) {
+  const input = new Tagify(el, {
+    autocomplete: true,
+    whitelist: get_recents(el.id) || [],
+    dropdown: {
+      enabled: 0
+    },
+  });
+
+  let abortController;
+  input.on('input', function(e) {
+    const value = e.detail.value;
+    input.settings.whitelist = []; // reset the whitelist
+
+    if (el.dataset.autocomplete && el.dataset.autocomplete !== "") {
+      // https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort
+      if (abortController) {
+        abortController.abort();
+      }
+
+      abortController = new AbortController();
+
+      fetch(el.dataset.autocomplete + 'term=' + value, {signal: abortController.signal}).
+        then((RES) => RES.json()).
+        then(function(whitelist) {
+          input.settings.whitelist = whitelist;
+          input.dropdown.show.call(input, value); // render the suggestions dropdown
+      });
+    }
+  });
+
+  input.on('add', function(e) {
+    let obj;
+
+    try {
+      obj = JSON.parse(window.localStorage.getItem("po_last_tags"));
+    } catch(err) {
+      if(err.name == "NS_ERROR_FILE_CORRUPTED") {
+          obj = null;
+      }
+    }
+
+    const tag = e.detail.data.value;
+    if (obj === null) {
+      obj = {};
+      obj[el.id] = [tag];
+    } else if (obj[el.id] === null) {
+      obj[el.id] = [tag];
+    } else {
+      if (obj[el.id].indexOf(tag) != -1) {
+        return;
+      }
+
+      if (obj[el.id].length >= maximumRecentEntriesPerTag) {
+        obj[el.id].pop();
+      }
+
+      obj[el.id].unshift(tag);
+    }
+
+
+    try {
+      window.localStorage.setItem("po_last_tags", JSON.stringify(obj));
+    } catch(err) {
+      if(err.name == "NS_ERROR_FILE_CORRUPTED") {
+        // TODO: Tell the user
```

Comment:
This shouldn't be left as a `TODO`.

Reply:
Let's just turn the feature off whenever there is an exception, without telling the user.